lib/internal/test_runner/test.js

```diff
@@ -762,6 +768,12 @@ class TestHook extends Test {
     super({ __proto__: null, fn, timeout, signal });
   }
   run(args) {
+    if (this.error) {
```

Comment:
this should only apply to hooks we store on the parent and we re-run for each child (`beforeEach` and `afterEach`)

Reply:
the bug exists also for hooks in a suite (see the added test)

and it is already only for `beforeEach` and `afterEach` as on first run `error` would be `null` and for `before` and `after` this won't get called again...